### PR TITLE
Feature/support mobile tracking

### PR DESCRIPTION
**Pull request by rlh1994**
## Description & motivation
This PR adds support for modelling mobile e-commerce events in the same package and runs as the web events. The changes aren't particularly complex but it has meant touching basically every test file. To make reviewing easier I've split it across a few commits, but there are some fixes in later commits to errors I made in the first few.

The basic idea is to just coalesce the mobile session id (from the context) and the screen view id (from the context) into the session and page view ids, everything else in the package then just works downstream from those. It has meant some annoying changes to the pg/rs events this run model, but it was the best way I could see to do it.

There is also a fix/improvement buried in here to get around the bigquery int test specific stuff, I did check you could still overwrite stuff locally, but might be worth double checking.

Note that this only adds mobile tracking if you already track on web, it is not possible to only model mobile events, mainly because I didn't think about it until right now. We can add this in a future PR if needed, the only thing you should have to do is not rely on the web_page_view context existing so should be even less changes if we do get a request for it.

Disabled databricks for now while it's still broken for actions, but it runs locally.

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have raised a [documentation](https://github.com/snowplow/documentation) PR if applicable (Will do in release PR)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 This dbt package:
 
-- Transforms and aggregates raw web e-commerce event data collected from the [Snowplow JavaScript tracker][tracker-docs] into a set of derived tables based around the following e-commerce data objects: carts, checkouts, products and transactions.
+- Transforms and aggregates raw web (and optionally mobile) e-commerce event data collected from the [Snowplow JavaScript tracker][tracker-docs] (and is/android Tracker) into a set of derived tables based around the following e-commerce data objects: carts, checkouts, products and transactions.
```
Comment (agnes-kiss) [inline README.md]:
```suggestion
- Transforms and aggregates raw web (and optionally mobile) e-commerce event data collected from the [Snowplow JavaScript tracker][tracker-docs] (and iOS/android Tracker) into a set of derived tables based around the following e-commerce data objects: carts, checkouts, products and transactions.
```

**Review by agnes-kiss — APPROVED** [review]
LGTM! before the description

**Comment by agnes-kiss** [comment]
branch is still main!

File: integration_tests/.scripts/integration_test.sh [hunk 2]
```diff
@@ -25,11 +25,15 @@ for db in ${DATABASES[@]}; do
...
+  eval "dbt seed --full-refresh --target $db" || exit 1;
+
+    echo "Snowplow e-commerce integration tests: Execute models (no mobile) - run 0/4"
+
+  eval "dbt run --full-refresh --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 30, snowplow__enable_mobile_events: false}' --target $db" || exit 1;
 
   echo "Snowplow e-commerce integration tests: Execute models - run 1/4"
 
-  eval "dbt run --target $db --full-refresh --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 30}'" || exit 1;
+  eval "dbt run --full-refresh --vars '{snowplow__allow_refresh: true, snowplow__backfill_limit_days: 30}' --target $db" || exit 1;
```
Comment (agnes-kiss) [inline integration_tests/.scripts/integration_test.sh:36]: 💜

File: models/base/scratch/bigquery/snowplow_ecommerce_base_events_this_run.sql [hunk 3]
```diff
@@ -21,61 +21,33 @@ from (
...
+          relation_alias='a',
+          include_field_alias=false) }},
+          a.contexts_com_snowplowanalytics_snowplow_web_page_1_0_0[safe_offset(0)].id
+        ) as page_view_id,
+        coalesce(
+          {{ snowplow_utils.get_optional_fields(
+          enabled=var('snowplow__enable_mobile_events', false),
+          fields=[{'field': 'session_id', 'dtype': 'string'}],
+          col_prefix='contexts_com_snowplowanalytics_snowplow_client_session_1_',
+          relation=source('atomic', 'events'),
```
Comment (agnes-kiss) [inline models/base/scratch/bigquery/snowplow_ecommerce_base_events_this_run.sql:40]: I tested overwriting the schema and tablename as well and worked fine as expected.

File: models/base/scratch/default/snowplow_ecommerce_base_events_this_run.sql [hunk 4]
```diff
@@ -179,128 +175,293 @@ with events_this_run AS (
...
+{%- endif %}
+
 {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__sde_ecommerce_action'), lower_limit, upper_limit, 'ecommerce_action', single_entity = false) }},
 
 {{ snowplow_utils.get_sde_or_context(var('snowplow__atomic_schema', 'atomic'), var('snowplow__context_web_page'), lower_limit, upper_limit, 'page_view') }}
 
+select
+    *,
+    dense_rank() over (partition by domain_sessionid order by derived_tstamp) AS event_in_session_index
+from (
```
Comment (agnes-kiss) [inline models/base/scratch/default/snowplow_ecommerce_base_events_this_run.sql]: The subquery here and removing the filter on the collector_tstamp earlier might cause performance issues. I'm tempted to suggest to keep a separate sql for the two situations until we find a less hectic solution.

Comment (rlh1994) [reply models/base/scratch/default/snowplow_ecommerce_base_events_this_run.sql]: Yeah I did try to keep it optimised, but also I looked back at it this morning and don't know why I did it in the first place, new version pushed.

**Review by agnes-kiss — APPROVED** [review]

File: models/base/scratch/default/snowplow_ecommerce_base_events_this_run.sql (hunk 4 above)
Comment (agnes-kiss) [reply models/base/scratch/default/snowplow_ecommerce_base_events_this_run.sql]: Much more elegant, nice!

**Pull request merged**